### ISSUE TITLE
loss return from SpeechToSpeechOutput cannot fit to Seq2SeqTrainer

### ISSUE THREAD

**Issue by indiejoseph** [huggingface/transformers#29276]
Seq2SeqTrainer cannot recognize the loss returned from SpeechToSpeechOutput, and return this error:

```
The model did not return a loss from the inputs, only the following keys: spectrum, encoder_last_hidden_state. For reference, the inputs it received are input_values, attention_mask, labels
```

I cannot find any code or tutorial about training SpeechT5SpeechToSpeech model, I can only found the training source code in fairseq's repo, but it is obsolete and not well documented how to prepare the training dataset.

**Comment by ArthurZucker** [comment]
cc @ylacombe and @sanchit-gandhi

**Comment by ylacombe** [comment]
Fixed by #29357
